src/main/java/io/cryostat/net/OpenShiftAuthManager.java

```diff
@@ -296,6 +311,26 @@ public AuthenticationScheme getScheme() {
         }
     }
 
+    private Future<Boolean> deleteToken(String token) {
+        try (OpenShiftClient client = clientProvider.apply(getServiceAccountToken())) {
+            Boolean deleted =
+                    Optional.ofNullable(
+                                    client.oAuthAccessTokens()
+                                            .withName(this.getOauthAccessTokenName(token))
+                                            .delete())
+                            .orElseThrow(TokenNotFoundException::new);
+
+            if (Boolean.FALSE.equals(deleted)) {
+                throw new TokenNotFoundException();
+            }
+
+            return CompletableFuture.completedFuture(deleted);
+        } catch (Exception e) {
+            logger.error(e);
+            return CompletableFuture.failedFuture(e);
```

Comment:
Since the `Future` returned here is always one that is created and populated with its value/exception immediately, and this is only an internal helper method, I think it can just be simplified to return a `boolean` rather than `Future<Boolean>`. It's all synchronous/single-threaded as it is, anyway, so it would simplify it down a little and at least lead to less `ExecutionException` wrapping/unwrapping in failure cases.

Reply:
I'll refactor it in this PR